src/Symfony/Component/Amqp/RetryStrategy/ConstantRetryStrategy.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Amqp\RetryStrategy;
+
+use Symfony\Component\Amqp\Exception\InvalidArgumentException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class ConstantRetryStrategy implements RetryStrategyInterface
+{
+    private $time;
+    private $max;
+
+    /**
+     * @param int $time Time to wait in the queue in seconds
+     * @param int $max  The maximum number of attempts (0 means no limit)
+     */
+    public function __construct(int $time, int $max = 0)
+    {
+        $time = (int) $time;
+
+        if ($time < 1) {
+            throw new InvalidArgumentException('"time" should be at least 1.');
```

Comment:
This looks like an arbitrary decision. I expected `0` to be allowed and mean "no wait".

Reply:
If you don't want to wait, there is no need to use a retry strategy. A simple nack + requeue is enough (I guess)